provisioning_templates/registration/linux_registration_default.erb

```diff
@@ -0,0 +1,67 @@
+<%#
+kind: registration
+name: Linux registration default
+model: ProvisioningTemplate
+oses:
+- CentOS
+- Fedora
+-%>
+#!/bin/bash
+
+set -e
+
+# This script enrolls a host with Foreman.
+# Host: <%= @host.name %>
+<% if @host.puppetmaster.present? -%>
+
+# Setup puppet repositories
+
+rpm -Uvh https://yum.puppetlabs.com/puppet6/puppet6-release-el-7.noarch.rpm
+
+# Install puppet
+
+if [ -f /usr/bin/dnf ]; then
+  dnf -y install puppet
+else
+  yum -t -y install puppet-agent
+fi
```

Comment:
This should use https://github.com/theforeman/community-templates/blob/develop/provisioning_templates/snippet/puppetlabs_repo.erb